### PR TITLE
Replace @job_queue with if-statements

### PR DESCRIPTION
**Pull request by dafeda**
**Approach**
Although I do like `DRY`, I'm not a fan of this decorator.
Proposing to replace it with some simple if-statements.

Also, I think `update_progress_for_index` should be `update_progress_for_iteration` since the parameter `idx` is never used.
Proposing this change as well.

## Pre review checklist

- [x ] Added appropriate labels

### CODE REVIEW

**Review by oyvindeide — APPROVED** [review]
Agree that this reads better 👍

**Review by oyvindeide — CHANGES_REQUESTED** [review]
Can you rename your second commit? Other than this looks good, the flakyness of the tests should be solved now.

**Review by oyvindeide — APPROVED** [review]
:rocket:

**Comment by oyvindeide** [comment]
Might be superseded by: https://github.com/equinor/ert/pull/3019

**Comment by dafeda** [comment]
test this please

**Comment by sondreso** [comment]
Needs a rebase after https://github.com/equinor/ert/pull/3019 was merged. Probably just drop the commit with the job_queue changes, as these were completely removed.

File: ert_shared/models/base_run_model.py [hunk 1]
```diff
@@ -186,11 +197,44 @@ def get_runtime(self) -> Union[int, float]:
...
+    def getQueueSize(self) -> int:
+        if self._job_queue is None:
+            return 1
+
+        queue_size = len(self._job_queue)
+
+        if queue_size == 0:
+            queue_size = 1
+
+        return queue_size
```
Comment (sondreso) [inline ert_shared/models/base_run_model.py:209]: Think something might have gone wrong in the rebase, as the functions are now reintroduced :thinking:

**Review by sondreso — CHANGES_REQUESTED** [review]

File: ert_shared/models/base_run_model.py (hunk 1 above)
Comment (dafeda) [reply ert_shared/models/base_run_model.py:209]: This PR does not make that much sense anyway after job_queue was removed.

**Pull request closed without merging**